File: 2696. Minimum String Length After Removing Substrings
/Solution.py
class Solution:
    def minLength(self, s: str) -> int:
        stack = []
        for c in s:
            if c == 'B':
                if stack and stack[-1] == 'A':
                    stack.pop()
                else:
                    stack.append(c)
            elif c == 'D':
                if stack and stack[-1] == 'C':
                    stack.pop()
                else:
                    stack.append(c)
            else:
                stack.append(c)
        return len(stack)

File: 2696. Minimum String Length After Removing Substrings
/Solution.cpp
class Solution {
public:
    int minLength(string s) {
        stack<char> stack;
        int n = s.length();
        for (int i = 0; i < n; i++) {
            if (s[i] == 'B') {
                if (!stack.empty() && stack.top() == 'A') {
                    stack.pop();
                } else {
                    stack.push(s[i]);
                }
            } else if (s[i] == 'D') {
                if (!stack.empty() && stack.top() == 'C') {
                    stack.pop();
                } else {
                    stack.push(s[i]);
                }
            } else {
                stack.push(s[i]);
            }
        }
        return stack.size();
    }
};
